tools/env-loader/cmd/env-loader_test.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"slices"
+	"testing"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseCli(t *testing.T) {
+	parseCLI([]string{})
+
+	flags := kingpin.CommandLine.Model().Flags
+	flagEnvVars := make([]string, 0, len(flags))
+	for _, flag := range flags {
+		if flag.Envar != "" {
+			flagEnvVars = append(flagEnvVars, flag.Envar)
+		}
+	}
+
+	uniqueFlagEnvVars := slices.Compact(slices.Clone(flagEnvVars))
+
+	require.ElementsMatch(t, flagEnvVars, uniqueFlagEnvVars, "not all flag env vars are unique")
+}
+
+func TestGetRequestedEnvValues(t *testing.T) {
+	tests := []struct {
+		desc           string
+		c              *config
+		expectedValues map[string]string
+	}{
+		{
+			desc: "specific values",
+			c: &config{
+				EnvironmentsDirectory: filepath.Join("..", "pkg", "testdata", "repos", "basic repo", ".environments"),
+				Environment:           "env1",
+				ValueSets: []string{
+					"testing1",
+				},
+				Values: []string{
+					"setLevel",
+					"envLevelCommon1",
+				},
+			},
+			expectedValues: map[string]string{
+				"setLevel":        "set level",
+				"envLevelCommon1": "env level",
+			},
+		},
+		{
+			desc: "full value set",
+			c: &config{
+				EnvironmentsDirectory: filepath.Join("..", "pkg", "testdata", "repos", "basic repo", ".environments"),
+				Environment:           "env1",
+				ValueSets: []string{
+					"testing1",
+				},
+			},
+			expectedValues: map[string]string{
+				"setLevel":        "set level",
+				"setLevelCommon":  "testing1 level",
+				"envLevelCommon1": "env level",
+				"envLevelCommon2": "set level",
+				"topLevelCommon1": "top level",
+				"topLevelCommon2": "env level",
+			},
+		},
+		{
+			desc: "specific env",
+			c: &config{
+				EnvironmentsDirectory: filepath.Join("..", "pkg", "testdata", "repos", "basic repo", ".environments"),
+				Environment:           "env1",
+			},
+			expectedValues: map[string]string{
+				"envLevelCommon1": "env level",
+				"envLevelCommon2": "env level",
+				"topLevelCommon1": "top level",
+				"topLevelCommon2": "env level",
+			},
+		},
+	}
+
+	for _, test := range tests {
+		actualValues, err := getRequestedEnvValues(test.c)
+		require.NoError(t, err)
+		require.EqualValues(t, actualValues, test.expectedValues)
```

Comment:
I think you've got the order backwards here. I'm pretty sure it should be expected values, then actual values.

In Teleport we have a linter enabled that catches a lot of these.

Reply:
> In Teleport we have a linter enabled that catches a lot of these.

Do you happen to know what it is/how it's configured? I setup [this](https://github.com/gravitational/shared-workflows/blob/main/tools/repo-release-tooling/tooling.mk#L25) to standardize how we lint across the projects in this repo. It'd be great to get this enabled for everything.

Reply:
We run `testifylint` via `golangci-lint`: https://github.com/gravitational/teleport/blob/master/.golangci.yml#L252-L264